editoast/src/views/mod.rs

```diff
@@ -442,6 +443,41 @@ mod tests {
         }};
     }
 
+    /// This is a helper function to create a test connection pool, is need to be created for each test
+    pub async fn create_test_connection_pool() -> DbConnectionPoolV2 {
+        let pg_config_url = PostgresConfig::default()
+            .url()
+            .expect("cannot get postgres config url");
+        DbConnectionPoolV2::create_from_url(pg_config_url, 10)
+            .await
+            .expect("cannot create test pool")
+    }
+
+    /// This is a helper function to create a test service with a connection pool in test mode, is need to be created for each test
+    pub async fn create_test_service_with_connection_pool(
+        connection_pool: DbConnectionPoolV2,
+    ) -> impl Service<Request, Response = ServiceResponse<BoxBody>, Error = Error> {
+        let redis = RedisClient::new(RedisConfig::default()).expect("cannot get redis client");
+
+        // Custom Json extractor configuration
+        let json_cfg = JsonConfig::default()
+            .limit(250 * 1024 * 1024) // 250MB
+            .error_handler(|err, _| InternalError::from(err).into());
+
+        let core: CoreClient = CoreClient::default();
```

Comment:
As discussed, maybe a builder would be a better idea. I suggest the following api, feel free to improve it:

```rust
// unused pool, default core client
let app = TestApp::default();

// explicitly specified pool and mocking client
let app = TestApp::new().pool(pool.clone()).core_client(my_mocking_client).build();

// implicitly created pool, default core client
let app = TestApp::default();
let pool = app.get_pool(); // When building the app, we could store a ref to the pool, so that we can access it later :)
```